### PR TITLE
netspeed: add netlink support

### PR DESCRIPTION
**Pull request by rbuj**
Based on https://stackoverflow.com/questions/18062268/using-nl80211-h-to-scan-access-points
and https://mirrors.edge.kernel.org/pub/software/network/iw/iw-5.4.tar.gz

### CODE REVIEW

**Comment by raveit65** [comment]
Should we start with this PR to review or with another one fir netspeed-applet ?
It adds support for libnl, does it remove libiw support?

File: netspeed/src/backend.c [hunk 1]
```diff
@@ -327,3 +350,413 @@ get_wireless_info (DevInfo *devinfo)
...
+    if (!devinfo->running)
+        goto cleanup;
+
+    /* Get in/out bitrate/rate/total, siganl quality from station message */
+    msg = nlmsg_alloc ();
+    if (!msg) {
+        g_warning ("failed to allocate netlink message");
+        goto cleanup;
+    }
+    genlmsg_put (msg, 0, 0, nlstate.nl80211_id, 0, NLM_F_DUMP, NL80211_CMD_GET_STATION, 0);
```
Comment (raveit65) [inline netspeed/src/backend.c:743]: There is a build warning. https://travis-ci.org/github/mate-desktop/mate-applets/jobs/705601948#L4769
```
backend.c: In function 'get_wireless_info':

backend.c:743:5: warning: '*((void *)&nlstate+8)' may be used uninitialized in this function [-Wmaybe-uninitialized]

  743 |     genlmsg_put (msg, 0, 0, nlstate.nl80211_id, 0, NLM_F_DUMP, NL80211_CMD_GET_STATION, 0);

      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: netspeed/src/backend.c [hunk 2]
```diff
@@ -327,3 +350,413 @@ get_wireless_info (DevInfo *devinfo)
...
+    if (tb_msg[NL80211_ATTR_WIPHY_FREQ]) {
+        uint32_t freq = nla_get_u32(tb_msg[NL80211_ATTR_WIPHY_FREQ]);
+        char buf[100];
+        char *pos = buf;
+
+        pos += sprintf(pos, _("%d (%d MHz)"),
+                       ieee80211_frequency_to_channel (freq), freq);
+
+        if (tb_msg[NL80211_ATTR_CHANNEL_WIDTH])
+            pos += sprintf(pos, _(", width: %s"),
```
Comment (raveit65) [inline netspeed/src/backend.c:662]: Build warning. https://travis-ci.org/github/mate-desktop/mate-applets/jobs/705601948#L4820
```
backend.c:662:13: warning: Value stored to 'pos' is never read

            pos += sprintf(pos, _(", width: %s"),

            ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by raveit65** [comment]
Very nice job.
netlink gives us much more connection informations.
with iwlib:
Netspped-applet-iwlib-old
with netlink
netspeed-applet-netlink-new

Btw. Can be done in another PR, but i think we should rename `internet address` to `IPV4 Address`
That a box has a direct connection to internet is very rare case nowadays, most boxes are in network behind a router or in public wifi hotspot.
A public internet address is something what you can get with:
```
$ curl ifconfig.me
158.18X.XX.XXX
```

File: netspeed/src/backend.c (hunk 1 above)
Comment (rbuj) [reply netspeed/src/backend.c:742]: done

File: netspeed/src/backend.c (hunk 2 above)
Comment (rbuj) [reply netspeed/src/backend.c]: done

File: netspeed/src/backend.c [hunk 3]
```diff
@@ -327,3 +350,412 @@ get_wireless_info (DevInfo *devinfo)
...
+void
+get_wireless_info (DevInfo *devinfo)
+{
+    struct nl80211_state nlstate = {.sock = NULL, .nl80211_id = -1};
+    int err, ret;
+    struct nl_msg *msg;
+    signed long long devidx = 0;
+
+    err = nl80211_init (&nlstate);
+    if (err || (nlstate.sock < 0)) {
```
Comment (raveit65) [inline netspeed/src/backend.c:680]: Now we got another build warning.
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/707239813#L4724
```
backend.c: In function 'get_wireless_info':

backend.c:680:30: warning: ordered comparison of pointer with integer zero [-Wextra]

  680 |     if (err || (nlstate.sock < 0)) {

      |                              ^

In file included from /usr/lib64/glib-2.0/include/glibconfig.h:9,

                 from /usr/include/glib-2.0/glib/gtypes.h:32,

                 from /usr/include/glib-2.0/glib/galloca.h:32,

                 from /usr/include/glib-2.0/glib.h:30,

                 from /usr/include/gtk-3.0/gdk/gdkconfig.h:13,

                 from /usr/include/gtk-3.0/gdk/gdk.h:30,

                 from /usr/include/gtk-3.0/gtk/gtk.h:30,

                 from netspeed.c:27:

netspeed.c: In function 'update_quality_icon':

/usr/include/glib-2.0/glib/gmacros.h:811:63: warning: comparison of unsigned expression in '< 0' is always false [-Wtype-limits]
```

Comment (rbuj) [reply netspeed/src/backend.c]: done

**Pull request merged**